scripts/populate-popular-and-new-and-noteworthy-projects.py

```
@@ -0,0 +1,116 @@
+
+    for contrib in node.contributors:
+        if contrib._id in qa_user_ids:
+            logger.warn('Node {} skipped because a QA member, {}, is a contributor.'.format(node._id, contrib._id))
+            return False
+
+    return True
+
+
+def update_node_links(designated_node, target_nodes, description):
+    """ Takes designated node, removes current node links and replaces them with node links to target nodes """
+    logger.warn('Repopulating {} with latest {} nodes.'.format(designated_node._id, description))
+    user = designated_node.creator
+    auth = Auth(user)
+
+    for i in xrange(len(designated_node.nodes)-1, -1, -1):
+        pointer = designated_node.nodes[i]
+        designated_node.rm_pointer(pointer, auth)
+
+    for n_id in target_nodes:
+        n = models.Node.find(Q('_id', 'eq', n_id))[0]
+        if is_eligible_node(n):
+            designated_node.add_pointer(n, auth, save=True)
+            logger.info('Added node link {} to {}'.format(n, designated_node))
+
+def main(dry_run=True):
+    init_app(routes=False)
+
+    with TokuTransaction():
+        # popular_nodes = get_popular_nodes()['popular_node_ids'] # TODO uncomment this
+        popular_links_node = models.Node.find(Q('_id', 'eq', POPULAR_LINKS_NODE))[0]
+        popular_nodes = ["dy68x", "zk45e", "acw57", "hucnr", "4e8zs"]  # TODO delete this
+        new_and_noteworthy_links_node = models.Node.find(Q('_id', 'eq', NEW_AND_NOTEWORTHY_LINKS_NODE))[0]
+        new_and_noteworthy_nodes = get_new_and_noteworthy_nodes()
+
+        update_node_links(popular_links_node, popular_nodes, 'popular')
+        update_node_links(new_and_noteworthy_links_node, new_and_noteworthy_nodes, 'new and noteworthy')
+
+        if not dry_run:
+            try:
+                popular_links_node.save()
+                logger.info('Node links on {} updated.'.format(popular_links_node._id))
+            except:
+                logger.error('Could not migrate popular nodes due to error')
```

Check out the `traceback` module in the standard library. For example, it has a `print_exc` method that could help here.

That said, I don't see that we're performing any additional actions following the logger.error call. Why not re-raise the error? And in that case, why not just let the error raise in the first place? Why trap and log it at all?